definitions/features/installer.rb

```diff
@@ -92,9 +92,7 @@ def run(arguments = '', exec_options = {})
   end
 
   def upgrade(exec_options = {})
-    arguments = '--disable-system-checks'
-    arguments += ' --upgrade' if can_upgrade?
-    run(arguments, exec_options)
+    run(installer_args, exec_options)
```

Comment:
Two places where we have used `--upgrade` flag. One inside upgrade method of installer feature and second passed as argument to `Procedures::Installer::Run` in package scenario.  

Adding a condition for `--upgrade` flag inside `upgrade` method and changing procedure from `Procedures::Installer::Run` to `Procedures::Installer::Upgrade` inside package scenario won't be sufficient here?

Reply:
I moved this as per the suggestion, changed tests accordingly. I am not sure the way I have handled the test change is correct or not, request to review. Thank you !